src/hooks/useClickSwitch.ts

```diff
@@ -0,0 +1,11 @@
+import { useState } from 'react';
+
+export const useClickSwitch = (def = true): [boolean, () => void] => {
+    const [on, setOn] = useState(def);
+
+    const onClickSwitch = () => {
+        setOn(!on);
+    };
```

Comment:
this may implements with using `useReducer`

`const [on, switch] = useReducer((s) => !s, def)`